### DISCUSSION TITLE
eglot does not seem to play nicely with desktop-save-mode

### DISCUSSION THREAD

**Discussion by petergardfjall** [joaotavora/eglot#1183] [Q&A] [answered]
Hi, and first of all thanks for the awesome work that you do!

I'm currently running `eglot` in `emacs-29` (https://github.com/emacs-mirror/emacs/commit/4b6f2a7028b91128934a19f83572f24106782225) and I use `desktop-save-mode`  to manage my buffers when developing. Recently I noticed that restoring a desktop fails with something like the output below.

I haven't seen this issue before so I assume that it has been introduced fairly recently (I try to stay pretty close to the latest commits on `emacs-29`). I can't say for certain though when this started happening.

Anyway, I wanted to report this, but didn't know how to go about it. 
Please let me know if I should take some other action.

```
Debugger entered--Lisp error: (jsonrpc-error "No current JSON-RPC connection" (jsonrpc-error-code . 32603) (jsonrpc-error-message . "No current JSON-RPC connection"))
  jsonrpc-error("No current JSON-RPC connection")
  eglot--current-server-or-lose()
  eglot--server-capable(:inlayHintProvider)
  eglot-inlay-hints-mode(1)
  desktop-create-buffer(208 "/home/peterg/dev/softsense/git/myproj/pkg/postgr..." "session_db_test.go" go-ts-mode (font-lock-mode undo-tree-mode corfu-mode eglot-inlay-hints-mode eldoc-mode flymake-mode eglot--managed-mode yas-minor-mode display-line-numbers-mode flyspell-mode diff-hl-mode) 19088 (19088 nil) nil nil ((indent-tabs-mode . t) (buffer-display-time 25619 3866 799578 177000) (buffer-file-coding-system . undecided-unix) (fill-column . 100)) ((mark-ring (19088 20422 20398 19780 17102 17524 19334 19780 18375))))
  load-with-code-conversion("/home/peterg/.emacs.d/wsp/myworkspace/desktop" "/home/peterg/.emacs.d/wsp/myworkspace/desktop" t t)
  load("/home/peterg/.emacs.d/wsp/myworkspace/desktop" t t t)
  desktop-read()
  wsp--desktop-activate("myworkspace")
  wsp--workspace-load("myworkspace")
  wsp-workspace-open("myworkspace")
  funcall-interactively(wsp-workspace-open "myproj")
  command-execute(wsp-workspace-open)
```

**Comment by joaotavora** [comment]
> Anyway, I wanted to report this, but didn't know how to go about it.
> Please let me know if I should take some other action.

Please, follow the steps in https://joaotavora.github.io/eglot/#Troubleshooting-Eglot, especially step 6.

Something is calling `eglot-inlay-hints-mode` when it shouldn't, that's all I can see from the backtrace.  But I can't see your configuration or figure out how to reproduce this, especially since I don't use desktop-save-mode myself.

**Reply by joaotavora** [reply]
FWIW I solved a simialr issue in https://debbugs.gnu.org/cgi/bugreport.cgi?bug=bug%2356407#68. Desktop mode messes with minor modes it doesn't own.  I think I know what the problem is.

**Reply by petergardfjall** [reply]
Thanks for responding so quickly! I'll see if I can get you something reproducable to work with!

> Something is calling `eglot-inlay-hints-mode` when it shouldn't

I'm not using `eglot-inlay-hints-mode` so that's a bit surprising.

**Reply by joaotavora** [reply]
> I'm not using eglot-inlay-hints-mode so that's a bit surprising.

It's on by default if the server supports Inlay Hints (and you haven't ignored that capability)

Fixed here: https://git.savannah.gnu.org/cgit/emacs.git/commit/?h=emacs-29&id=7385c991dff3466b37cf50628e7685cd53e71921

**Comment by joaotavora** [comment, marked as answer]
Fix pushed to emacs-29

**Reply by petergardfjall** [reply]
Wow, that was extremely quick! :star: I'll try this out as soon as the commit makes it over to the GitHub mirror!

**Reply by petergardfjall** [reply]
I verified and https://github.com/emacs-mirror/emacs/commit/7385c991dff3466b37cf50628e7685cd53e71921 fixes the issue!

Thank you so much! :star:
